services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/grpc/LogSpec.groovy

```diff
@@ -0,0 +1,114 @@
+package org.openkilda.functionaltests.spec.grpc
+
+import static org.junit.Assert.assertTrue
+import static org.openkilda.testing.ConstantsGrpc.DEFAULT_LOG_MESSAGES_STATE
+import static org.openkilda.testing.ConstantsGrpc.DEFAULT_LOG_OF_MESSAGES_STATE
+import static org.openkilda.testing.ConstantsGrpc.DEFAULT_REMOTE_LOG_IP
+import static org.openkilda.testing.ConstantsGrpc.DEFAULT_REMOTE_LOG_PORT
+import static org.openkilda.testing.ConstantsGrpc.REMOTE_LOG_IP
+import static org.openkilda.testing.ConstantsGrpc.REMOTE_LOG_PORT
+import static org.openkilda.testing.ConstantsGrpc.SWITCH
+
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.grpc.speaker.model.LogMessagesDto
+import org.openkilda.grpc.speaker.model.LogOferrorsDto
+import org.openkilda.grpc.speaker.model.RemoteLogServerDto
+import org.openkilda.messaging.error.MessageError
+import org.openkilda.messaging.model.grpc.OnOffState
+
+import groovy.util.logging.Slf4j
+import org.springframework.http.HttpStatus
+import org.springframework.web.client.HttpClientErrorException
+import spock.lang.Narrative
+import spock.lang.Unroll
+
+@Slf4j
+@Narrative("Test Log related stuff")
+class LogSpec extends BaseSpecification {
+    def "Able to enable log messages."() {
+        given: "Current state of log messages."
+        def states = getNewAndOriginalState(DEFAULT_LOG_MESSAGES_STATE)
+
+        when: "Try to apply a ${states[0]} state."
+        def r1 = grpc.enableLogMessagesOnSwitch(SWITCH, new LogMessagesDto(states[0]))
+
+        then: "The ${states[0]} state is applied."
+        assert r1.enabled == states[0]
+
+        when: "Restore original state."
+        def r2 = grpc.enableLogMessagesOnSwitch(SWITCH, new LogMessagesDto(states[1]))
+
+        then: "Original state os restored."
+        assert r2.enabled == states[1]
+    }
+
+    def "Able to enable OF log messages."() {
+        given: "Current state of OF log messages."
+        def states = getNewAndOriginalState(DEFAULT_LOG_OF_MESSAGES_STATE)
+
+        when: "Try to apply a ${states[0]} state."
+        def r1 = grpc.enableLogOfErrorsOnSwitch(SWITCH, new LogOferrorsDto(states[0]))
+
+        then: "The ${states[0]} state is applied."
+        assert r1.enabled == states[0]
+
+        when: "Restore original state."
+        def r2 = grpc.enableLogOfErrorsOnSwitch(SWITCH, new LogOferrorsDto(states[1]))
+
+        then: "Original state is restored."
+        assert r2.enabled == states[1]
+    }
+
+    def "Able to manipulate(CRUD) with a remote log server"() {
+        given: "Remove current remote log server configuration."
```

Comment:
Do you mind organizing the deletion process as another when-then pair, instead of making it 'given'?